File: api-cmssignature/csharp/test/src/test.esya.api.cmssignature/conversion/T01_BES.cs
using System;
using System.Collections.Generic;
using NUnit.Framework;
using test.esya.api.cmssignature.validation;
using tr.gov.tubitak.uekae.esya.api.cmssignature.attribute;
using tr.gov.tubitak.uekae.esya.api.cmssignature.signature;
using tr.gov.tubitak.uekae.esya.api.crypto.alg;
using tr.gov.tubitak.uekae.esya.asn.util;

namespace test.esya.api.cmssignature.conversion
{
    [TestFixture]
    public class T01_BES : CMSSignatureTest
    {
        private readonly String DIRECTORY = "T:\\api-cmssignature\\test-output\\csharp\\ma3\\conversion\\";

        [Test]
        public void testCreateBES()
        {
            BaseSignedData bs = new BaseSignedData();

            //add content which will be signed
            bs.addContent(getSimpleContent());

            //create parameters necessary for signature creation
            Dictionary<String, Object> parameters = new Dictionary<String, Object>();

            //necessary for certificate validation.By default,certificate validation is done.But if the user 
            //does not want certificate validation,he can add P_VALIDATE_CERTIFICATE_BEFORE_SIGNING parameter with its value set to false
            parameters[EParameters.P_CERT_VALIDATION_POLICY] = getPolicy();

            //add signer
            bs.addSigner(ESignatureType.TYPE_BES, getSignerCertificate(), getSignerInterface(SignatureAlg.RSA_SHA1),
                          null, parameters);

            //write the contentinfo to file
            AsnIO.dosyayaz(bs.getEncoded(), DIRECTORY + "BES.p7s");

            ValidationUtil.checkSignatureIsValid(bs.getEncoded(), null);
        }

        [Test]
        public void testCreateP1()
        {
            BaseSignedData bs = new BaseSignedData();

            //add content which will be signed
            bs.addContent(getSimpleContent());

            //Since SigningTime attribute is optional,add it to optional attributes list
            List<IAttribute> optionalAttributes = new List<IAttribute>();
            optionalAttributes.Add(new SigningTimeAttr(DateTime.UtcNow));

            //create parameters necessary for signature creation
            Dictionary<String, Object> parameters = new Dictionary<String, Object>();

            //necessary for certificate validation.By default,certificate validation is done.But if the user 
            //does not want certificate validation,he can add P_VALIDATE_CERTIFICATE_BEFORE_SIGNING parameter with its value set to false
            parameters[EParameters.P_CERT_VALIDATION_POLICY] = getPolicy();

            //add signer
            bs.addSigner(ESignatureType.TYPE_BES, getSignerCertificate(), getSignerInterface(SignatureAlg.RSA_SHA1),
                         optionalAttributes, parameters);

            //write the contentinfo to file
            AsnIO.dosyayaz(bs.getEncoded(), DIRECTORY + "EPES.p7s");

            ValidationUtil.checkSignatureIsValid(bs.getEncoded(), null);
        }
    }
}

File: api-cmssignature/testsrc/test/esya/api/cmssignature/convertion/T01_BES.java
package test.esya.api.cmssignature.convertion;

import org.junit.Test;
import test.esya.api.cmssignature.CMSSignatureTest;
import test.esya.api.cmssignature.validation.ValidationUtil;
import tr.gov.tubitak.uekae.esya.api.cmssignature.attribute.EParameters;
import tr.gov.tubitak.uekae.esya.api.cmssignature.attribute.IAttribute;
import tr.gov.tubitak.uekae.esya.api.cmssignature.attribute.SigningTimeAttr;
import tr.gov.tubitak.uekae.esya.api.cmssignature.signature.BaseSignedData;
import tr.gov.tubitak.uekae.esya.api.cmssignature.signature.ESignatureType;
import tr.gov.tubitak.uekae.esya.api.crypto.alg.SignatureAlg;
import tr.gov.tubitak.uekae.esya.asn.util.AsnIO;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.HashMap;
import java.util.List;

public class T01_BES extends CMSSignatureTest
{
	private final String DIRECTORY = "T:\\api-cmssignature\\test-output\\java\\ma3\\conversion\\";
	
	@Test
	public void testCreateBES() throws Exception
	{
		BaseSignedData bs = new BaseSignedData();
		bs.addContent(getSimpleContent());

		//create parameters necessary for signature creation
		HashMap<String, Object> params = new HashMap<String, Object>();
		params.put(EParameters.P_CERT_VALIDATION_POLICY, getPolicy());

		bs.addSigner(ESignatureType.TYPE_BES, getSignerCertificate(), getSignerInterface(SignatureAlg.RSA_SHA1), null, params);

		AsnIO.dosyayaz(bs.getEncoded(), DIRECTORY+"BES.p7s");

		ValidationUtil.checkSignatureIsValid(bs.getEncoded(), null);
	}

	@Test
	public void testCreateP1() throws Exception
	{
		BaseSignedData bs = new BaseSignedData();
		bs.addContent(getSimpleContent());

		List<IAttribute> optionalAttributes = new ArrayList<IAttribute>();
		optionalAttributes.add(new SigningTimeAttr(Calendar.getInstance()));

		HashMap<String, Object> params = new HashMap<String, Object>();
		params.put(EParameters.P_CERT_VALIDATION_POLICY, getPolicy());

		bs.addSigner(ESignatureType.TYPE_BES, getSignerCertificate() , getSignerInterface(SignatureAlg.RSA_SHA256), optionalAttributes, params);

		AsnIO.dosyayaz(bs.getEncoded(),DIRECTORY + "EPES.p7s");

		ValidationUtil.checkSignatureIsValid(bs.getEncoded(), null);
	}
}
